tests/measurements/test_counts.py

```diff
@@ -396,6 +396,32 @@ def test_counts_binsize(self):
 class TestCountsIntegration:
     # pylint:disable=too-many-public-methods,not-an-iterable
 
+    def test_counts_all_outcomes_with_mcm(self):
+        n_sample = 10
+
+        dev = qml.device("default.qubit", shots=n_sample)
+
+        @qml.qnode(device=dev, mcm_method="one-shot")
+        def single_mcm():
+            m = qml.measure(0)
+            return qml.counts(m, all_outcomes=True)
+
+        res = single_mcm()
+
+        assert list(res.keys()) == [0.0, 1.0]
```

Comment:
Is there any way we can have the key list dynamically generated but another dev? I'm thinking if so then when someday the '0.0, 1.0' thing fixed to normal '0 1' we don't have to modify this once again